src/activity_handlers.cpp

```diff
@@ -1486,7 +1486,7 @@ void activity_handlers::repair_item_finish( player_activity *act, player *p )
 
     // TODO: Allow setting this in the actor
     // TODO: Don't use charges_to_use: welder has 50 charges per use, soldering iron has 1
-    const int charges_to_use = used_tool->type->charges_to_use();
+    const int charges_to_use = used_tool->type->charges_to_use() + actor->charges_to_use;
```

Comment:
`charges_to_use` seems to be misleading, it's more like "*additional* charges to use". This can be become quite complicated as all checks now need to consider the new data from the actor *and* the generic data from `it_tool`. There should only be one value per item action.